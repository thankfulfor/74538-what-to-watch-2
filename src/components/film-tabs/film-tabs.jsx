import * as React from 'react';
import PropTypes from 'prop-types';
import {tabNames} from '../../utils/constants.js';

export const FilmTabs = (props) => {
  const {onTabClick, tabToShow} = props;

  return (
    <ul className="movie-nav__list">
      {tabNames.map((tabName) => (
        <li key={tabName} onClick={onTabClick} className={`movie-nav__item ${tabToShow === tabName && `movie-nav__item--active`}`}>
          <span className="movie-nav__link">{tabName}</span>
        </li>
      ))}
    </ul>
  );
};

FilmTabs.propTypes = {
  onTabClick: PropTypes.func.isRequired,
  tabToShow: PropTypes.string.isRequired,
};
