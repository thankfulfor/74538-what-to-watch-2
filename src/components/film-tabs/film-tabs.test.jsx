import React from 'react';
import renderer from 'react-test-renderer';
import {FilmTabs} from './film-tabs.jsx';
import {BrowserRouter} from 'react-router-dom';

it(`FilmTabs корректно рендерится после перезапуска`, () => {
  const clickHandler = jest.fn();
  const tree = renderer
    .create(
        <BrowserRouter>
          <FilmTabs
            tabToShow=''
            onTabClick={clickHandler}
          />
        </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
