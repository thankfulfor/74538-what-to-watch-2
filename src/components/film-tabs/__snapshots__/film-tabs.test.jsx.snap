// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmTabs корректно рендерится после перезапуска 1`] = `
<ul
  className="movie-nav__list"
>
  <li
    className="movie-nav__item false"
    onClick={[MockFunction]}
  >
    <a
      className="movie-nav__link"
      href="#"
    >
      Overview
    </a>
  </li>
  <li
    className="movie-nav__item false"
    onClick={[MockFunction]}
  >
    <a
      className="movie-nav__link"
      href="#"
    >
      Details
    </a>
  </li>
  <li
    className="movie-nav__item false"
    onClick={[MockFunction]}
  >
    <a
      className="movie-nav__link"
      href="#"
    >
      Reviews
    </a>
  </li>
</ul>
`;
